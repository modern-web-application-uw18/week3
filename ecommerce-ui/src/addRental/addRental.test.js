import React from 'react';
import ReactDOM from 'react-dom';
import AddRental from './addRental';
import TestRenderer from 'react-test-renderer'

describe("AddRental", () => {
    it('Smoke test make sure the component rendors', () =>{
        const div = document.createElement('div');
        ReactDOM.render(<AddRental />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Shoudl render a component with rental info and an add to cart button', () => {
        //create a snapshot
        const component = TestRenderer.create(<AddRental />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

