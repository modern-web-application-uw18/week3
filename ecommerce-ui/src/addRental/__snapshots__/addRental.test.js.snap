// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRental Shoudl render a component with rental info and an add to cart button 1`] = `
<div
  className="flex-item"
>
  <img
    alt=""
    height="164px"
    src={undefined}
    width="246px"
  />
  <p />
  <p />
  <p>
    , 
  </p>
  <p>
    $
     
  </p>
  <p>
    Stars: 
     Number of Reviews: 
  </p>
  <p>
    Host: 
     
  </p>
  <button
    onClick={[Function]}
  >
    Add Rental
  </button>
</div>
`;
